source/slang/slang-ir-use-uninitialized-values.cpp

```diff
@@ -348,8 +348,135 @@ namespace Slang
         return loads;
     }
 
+    static bool isInstStoredInto(ReachabilityContext& reachability, IRInst* reference, IRInst* inst)
+    {
+        auto addresses = getAliasableInstructions(inst);
+        
+        List<IRInst*> stores;
+        List<IRInst*> loads;
+
+        for (auto alias : addresses)
+        {
+            for (auto use = alias->firstUse; use; use = use->nextUse)
+            {
+                IRInst* user = use->getUser();
+                collectLoadStore(stores, loads, user, alias);
+            }
+        }
+
+        for (auto store : stores)
+        {
+            if (reachability.isInstReachable(store, reference))
+                return true;
+        }
+
+        return false;
+    }
+
+    static IRInst* traceInstOrigin(IRInst* inst)
+    {
+        if (auto load = as<IRLoad>(inst))
+            return traceInstOrigin(load->getPtr());
+
+        return inst;
+    }
+
+    static bool isFieldUsed(IRStructField* field, const List<IRStructKey*>& used)
+    {
+        for (auto key : used)
+        {
+            if (field->getKey() == key)
+                return true;
+        }
+
+        return false;
+    }
+
+    static List<IRStructField*> checkFieldsFromExit(ReachabilityContext& reachability, IRReturn* ret, IRStructType* type)
+    {
+        IRInst* origin = traceInstOrigin(ret->getVal());
+
+        // We don't want to warn on delegated construction
+        if (!isUninitializedValue(origin))
+            return {};
+
+        // Now we can look for all references to fields
+        List<IRStructKey*> usedKeys;
```

Comment:
You may want to use HashSet type for this variable.
If the user has a struct with hundreds members, it can make a big difference.

Reply:
Thanks, I was looking for a data structure to use here instead.